src/components/FacetTabs.js

```diff
@@ -0,0 +1,153 @@
+  static contextTypes = {
+    searcher: PropTypes.any,
+  };
+
+  static defaultProps = {
+    facetName: 'table',
+    facet: null,
+    backgroundColor: '#add8e6',
+  };
+
+  static displayName = 'FacetTabs';
+
+  constructor(props: FacetTabsProps) {
+    super(props);
+    this.state = {
+      tabsList: [],
+    };
+
+    (this: any).getTabsFromFacet = this.getTabsFromFacet.bind(this);
+    (this: any).populateTabs = this.populateTabs.bind(this);
+    (this: any).handleTabClick = this.handleTabClick.bind(this);
+  }
+
+  componentWillMount() {
+    this.populateTabs(this.props);
+  }
+
+  componentWillReceiveProps(newProps: FacetTabsProps) {
+    this.populateTabs(newProps);
+  }
+
+  //eslint-disable-next-line
+  numberWithCommas(x) {
+    return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+  }
+
+  handleTabClick(bucket: SearchFacetBucket, label: string) {
+    this.context.searcher.addFacetFilter(label, bucket.value, bucket.filter);
+  }
+
+  getTabsFromFacet(facet: SearchFacet) {
+    const tabs = [];
+    facet.buckets.forEach((bucket: SearchFacetBucket) => {
+      const value = bucket.value;
+      const count = this.numberWithCommas(bucket.count);
+      if (this.props.handleClick) {
+        tabs.push(
+          <div //eslint-disable-line
+            onClick={() => {
+              this.props.handleClick(bucket);
+            }}
+            className="facet-tab"
+          >
+            <div className="facet-tab-text">
+              {value} <span style={{ color: 'gray' }}>({count})</span>
+            </div>
+          </div>,
+        );
+      } else {
+        tabs.push(
+          <div //eslint-disable-line
+            onClick={() => {
+              this.handleTabClick(bucket, facet.label);
+            }}
+            className="facet-tab"
+          >
+            <div className="facet-tab-text">
+              {value} <span style={{ color: 'gray' }}>({count})</span>
+            </div>
+          </div>,
+        );
+      }
+    });
+    this.setState({ tabsList: tabs });
+  }
+
+  populateTabs(props: FacetTabsProps) {
+    const { facetName, facet } = props;
+
+    if (facet && facet.buckets) {
+      this.getTabsFromFacet(facet);
+    } else if (facetName) {
+      const searcher = this.context.searcher;
+      if (searcher && searcher.state.response) {
+        if (searcher.state.response.facets) {
+          searcher.state.response.facets.forEach((currentFacet: SearchFacet) => {
+            if (currentFacet.field === facetName && currentFacet.buckets) {
+              this.getTabsFromFacet(currentFacet);
+            }
+          });
+        }
+      }
+    }
+  }
+
+  render() {
+    const tabData = this.state.tabsList;
+    const tabs =
+      tabData.length > 0 ? (
+        <div
+          className="facet-tab-container"
+          style={{ backgroundColor: `${this.props.backgroundColor}` }}
+        >
+          {tabData}
+        </div>
+      ) : (
+        <BusyIndicator
```

Comment:
What if there are just no buckets for the facet? I don't think we want to show the spinny in that case...